lambda/facia-responder/src/main.ts

```diff
@@ -37,28 +50,58 @@ export const handler: SQSHandler = async (event) => {
 
 		//If something fails here, let it crash. The message will get retried and then sent to DLQ
 		// by the Lambda runtime and we will continue running
-		const newCuration = parseMesssage(rec);
+		const messageBody = getMessageBodyAsObject(rec);
+
+		// We parse the message in two phases, first extracting the message envelope,
+		// and then the fronts data. In this way, if extracting the fronts data fails,
+		// we can use the envelope data to post the error message back to the Fronts tool.
+		const maybeMessageEnvelope = parseFeastCurationEnvelope(messageBody);
+		if (!maybeMessageEnvelope.success) {
+			throw new Error(
+				`Error parsing message envelope: ${JSON.stringify(
+					maybeMessageEnvelope.error,
+				)}`,
+			);
+		}
+
+		const { edition, issueDate, version } = maybeMessageEnvelope.data;
+
+		const maybeFronts = parseFeastCuration(messageBody);
+
+		if (!maybeFronts.success) {
+			console.error(maybeFronts.error);
+			return notifyFaciaTool({
+				edition,
+				issueDate,
+				version,
+				status: 'Failed',
+				message: `Failed to publish this issue. Error: ${JSON.stringify(
+					maybeFronts.error,
+				)}`,
+				timestamp: Date.now(),
+			});
+		}
 
 		try {
-			await deployCuration(newCuration);
+			await deployCuration(maybeFronts.data);
+
+			return notifyFaciaTool({
+				edition,
+				issueDate,
+				version,
+				status: 'Published',
+				message: 'This issue has been published',
+				timestamp: Date.now(),
+			});
 		} catch (e) {
 			return notifyFaciaTool({
```

Comment:
Do we want to log the error out here too do you think?

Reply:
Good shout! [701dc99](https://github.com/guardian/recipes-backend/pull/54/commits/701dc99798803616bc54044646c03aeaacb23e8b)